### ISSUE TITLE
`mamba env create --dry-run` and `mamba env create` resolve different python version

### ISSUE THREAD

**Issue by ThomasBlauthQC** [mamba-org/mamba#1275]
I noticed that `mamba env create --dry-run` and `mamba env create` resolve different python versions (3.9.7 vs. 3.10.0) when creating a new environment only containing flake8. Is this intentional? Below is the output produced by both commands.

```console
$ cat requirements.txt
flake8

$ mamba env create -f requirements.txt -n flake8 --dry-run
Collecting package metadata (repodata.json): done
Solving environment: done
name: flake8
dependencies:
  - conda-forge/osx-64::ca-certificates==2021.10.8=h033912b_0
  - conda-forge/osx-64::libffi==3.4.2=h0d85af4_5
  - conda-forge/osx-64::libzlib==1.2.11=h9173be1_1013
  - conda-forge/osx-64::ncurses==6.2=h2e338ed_4
  - conda-forge/noarch::tzdata==2021e=he74cb21_0
  - conda-forge/osx-64::xz==5.2.5=haf1e3a3_1
  - conda-forge/osx-64::openssl==3.0.0=h0d85af4_2
  - conda-forge/osx-64::readline==8.1=h05e3726_0
  - conda-forge/osx-64::zlib==1.2.11=h9173be1_1013
  - conda-forge/osx-64::sqlite==3.36.0=h23a322b_2
  - conda-forge/osx-64::tk==8.6.11=h5dbffcc_1
  - conda-forge/osx-64::python==3.9.7=h38b4d05_3_cpython
  - conda-forge/noarch::mccabe==0.6.1=py_1
  - conda-forge/noarch::pycodestyle==2.8.0=pyhd8ed1ab_0
  - conda-forge/noarch::pyflakes==2.4.0=pyhd8ed1ab_0
  - conda-forge/osx-64::python_abi==3.9=2_cp39
  - conda-forge/noarch::wheel==0.37.0=pyhd8ed1ab_1
  - conda-forge/noarch::zipp==3.6.0=pyhd8ed1ab_0
  - conda-forge/osx-64::importlib-metadata==4.2.0=py39h6e9494a_0
  - conda-forge/osx-64::setuptools==59.1.1=py39h6e9494a_0
  - conda-forge/noarch::flake8==4.0.1=pyhd8ed1ab_1
  - conda-forge/noarch::pip==21.3.1=pyhd8ed1ab_0

$ mamba env create -f requirements.txt -n flake8
conda-forge/osx-64       [====================] (00m:00s) No change
conda-forge/noarch       [====================] (00m:00s) No change


Looking for: ['flake8']


Transaction

  Prefix: /usr/local/mambaforge/envs/flake8

  Updating specs:

   - flake8


  Package                 Version  Build               Channel                  Size
──────────────────────────────────────────────────────────────────────────────────────
  Install:
──────────────────────────────────────────────────────────────────────────────────────

  + bzip2                   1.0.8  h0d85af4_4          conda-forge/osx-64     Cached
  + ca-certificates     2021.10.8  h033912b_0          conda-forge/osx-64     Cached
  + flake8                  4.0.1  pyhd8ed1ab_1        conda-forge/noarch     Cached
  + importlib-metadata      3.3.0  pyhd8ed1ab_1        conda-forge/noarch     Cached
  + libffi                  3.4.2  h0d85af4_5          conda-forge/osx-64     Cached
  + libzlib                1.2.11  h9173be1_1013       conda-forge/osx-64     Cached
  + mccabe                  0.6.1  py_1                conda-forge/noarch     Cached
  + ncurses                   6.2  h2e338ed_4          conda-forge/osx-64     Cached
  + openssl                 3.0.0  h0d85af4_2          conda-forge/osx-64     Cached
  + pip                    21.3.1  pyhd8ed1ab_0        conda-forge/noarch     Cached
  + pycodestyle             2.8.0  pyhd8ed1ab_0        conda-forge/noarch     Cached
  + pyflakes                2.4.0  pyhd8ed1ab_0        conda-forge/noarch     Cached
  + python                 3.10.0  h38b4d05_2_cpython  conda-forge/osx-64     Cached
  + python_abi               3.10  2_cp310             conda-forge/osx-64     Cached
  + readline                  8.1  h05e3726_0          conda-forge/osx-64     Cached
  + setuptools             59.1.1  py310h2ec42d9_0     conda-forge/osx-64     Cached
  + sqlite                 3.36.0  h23a322b_2          conda-forge/osx-64     Cached
  + tk                     8.6.11  h5dbffcc_1          conda-forge/osx-64     Cached
  + typing_extensions       4.0.0  pyha770c72_0        conda-forge/noarch     Cached
  + tzdata                  2021e  he74cb21_0          conda-forge/noarch     Cached
  + wheel                  0.37.0  pyhd8ed1ab_1        conda-forge/noarch     Cached
  + xz                      5.2.5  haf1e3a3_1          conda-forge/osx-64     Cached
  + zipp                    3.6.0  pyhd8ed1ab_0        conda-forge/noarch     Cached
  + zlib                   1.2.11  h9173be1_1013       conda-forge/osx-64     Cached

  Summary:

  Install: 24 packages

  Total download: 0  B

──────────────────────────────────────────────────────────────────────────────────────

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate flake8
#
# To deactivate an active environment, use
#
#     $ conda deactivate

```

**Comment by wolfv** [comment]
Indeed, it looks like this is using the conda resolver in the dry-run case. 

Unfortunately the `env` subcommand code from conda is really messy ... `dry_run` was only added in April of this year. 
Not sure if we are going to implement this or just suggest everyone to switch to micromamba :)

**Comment by wolfv** [comment]
fixed in #1315
